src/test/operators/join_test.hpp

```diff
@@ -116,6 +117,59 @@ class JoinTest : public BaseTest {
     const auto actual_result = join->get_output();
     EXPECT_TABLE_EQ_UNORDERED(actual_result, expected_result);
 
+    /**
+             * Test the column definitions of the output table, especially the nullability
+             */
+
+    for (auto output_column_id = ColumnID{0}; output_column_id < actual_result->column_count(); ++output_column_id) {
+      auto expected_column_definition = TableColumnDefinition{};
+
+      switch (mode) {
+        case JoinMode::Inner:
+        case JoinMode::Left:
+        case JoinMode::Right:
+        case JoinMode::Outer:
+        case JoinMode::Cross:
+          if (output_column_id < left->get_output()->column_count()) {
+            expected_column_definition = left->get_output()->column_definitions()[output_column_id];
+            if (mode == JoinMode::Right || mode == JoinMode::Outer) expected_column_definition.nullable = true;
+          } else {
+            expected_column_definition =
+                right->get_output()->column_definitions()[output_column_id - left->get_output()->column_count()];
+            if (mode == JoinMode::Left || mode == JoinMode::Outer) expected_column_definition.nullable = true;
+          }
+          break;
+
+        case JoinMode::Semi:
+          expected_column_definition = left->get_output()->column_definitions()[output_column_id];
+          break;
+        case JoinMode::Anti:
+          expected_column_definition = right->get_output()->column_definitions()[output_column_id];
+          break;
+      }
+      const auto actual_column_definition = actual_result->column_definitions()[output_column_id];
+      EXPECT_EQ(actual_column_definition, expected_column_definition);
+    }
+  }
```

Comment:
It would be great to also check whether the number of columns matches the expected number.

Reply:
This now duplicates the code to check the column definitions in `test_hash_join_output`.